float64 distance
float64 speed
float64 force
bool release
bool enable

